src/Autocomplete/src/Calculator/ChecksumCalculator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symfony\UX\Autocomplete\Calculator;
+
+final class ChecksumCalculator implements ChecksumCalculatorInterface
```

Comment:
In my own selection type I did instead of using a checksum go with the `UriSigner` but ignore all query parameters which I don't know, so maybe we could do here the avoid own checksum calculator:

```php
if (!$this->uriSigner->check(
    // stripped down uri from the request
    $request->getSchemeAndHttpHost() . $request->getPathInfo() . '?_config=' . \rawurlencode($request->query->get('_config')) . '&_hash=' . \rawurlencode($request->query->get('_hash')),
)) {
    throw new AccessDeniedHttpException('Invalid signature.');
}

/**
 * @var array{
 *     options: array<string, mixed>,
 *     name: string,
 *     theme: string,
 * }|false
 */
$config = \json_decode(\base64_decode($request->query->get('_config'), true), true, 512, \JSON_THROW_ON_ERROR);
```